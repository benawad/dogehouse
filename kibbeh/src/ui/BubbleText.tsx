import * as React from "react";

export interface BubbleTextProps {
  live?: boolean;
  children: React.ReactNode;
}

export const BubbleText: React.FC<BubbleTextProps> = ({ live, children }) => {
  return (
    <div
      className="text-primary-200 font-bold items-center"
      data-testid="bubble-text"
    >
      <div
        className={`inline-block mr-2 w-2 h-2 rounded-full ${
          live ? "bg-accent" : "bg-primary-300"
        }`}
      ></div>
      {children}
    </div>
  );
};
