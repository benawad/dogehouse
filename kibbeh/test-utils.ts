import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import "mutationobserver-shim";

const Renderer = ({ children }) => {
  return { children };
};

const customRender = (ui, options = {}) =>
  render(ui, {
    ...options,
  });

export * from "@testing-library/react";

export { customRender as render };
