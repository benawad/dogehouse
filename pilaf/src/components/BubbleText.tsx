import React from "react";
import { StyleSheet, View, ViewStyle } from "react-native";
import { colors } from "../constants/dogeStyle";

export interface BubbleTextProps {
  style?: ViewStyle;
  live?: boolean;
  children: React.ReactNode;
}

export const BubbleText: React.FC<BubbleTextProps> = ({
  live,
  children,
  style,
}) => {
  return (
    <View style={[styles.container, style]}>
      <View
        style={[
          styles.liveDot,
          live
            ? { backgroundColor: colors.accent }
            : { backgroundColor: colors.primary300 },
        ]}
      />
      {children}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    alignItems: "center",
  },
  liveDot: {
    height: 8,
    width: 8,
    borderRadius: 4,
    marginRight: 4,
  },
});
