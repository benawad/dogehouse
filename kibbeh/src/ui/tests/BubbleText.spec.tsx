import React from "react";

import { render } from "../../../test-utils";
import { BubbleText } from "../BubbleText";

describe("BubbleText", () => {
  it("should render correctly", () => {
    const { getByTestId, getByText } = render(<BubbleText>Child</BubbleText>);
    const component = getByTestId("bubble-text");
    const child = getByText("Child");

    expect(component).toBeVisible();
    expect(child).toBeVisible();
  });

  it("should have bg-accent class if live prop is true", () => {
    const { getByTestId } = render(<BubbleText live={true}>Child</BubbleText>);
    const component = getByTestId("bubble-text");

    expect(
      (component.firstChild as HTMLDListElement).classList.contains("bg-accent")
    ).toBeTruthy();
    expect(
      (component.firstChild as HTMLDListElement).classList.contains(
        "bg-primary-300"
      )
    ).toBeFalsy();
  });

  it("should have bg-primary-300 class if live prop is false", () => {
    const { getByTestId } = render(<BubbleText live={false}>Child</BubbleText>);
    const component = getByTestId("bubble-text");

    expect(
      (component.firstChild as HTMLDListElement).classList.contains("bg-accent")
    ).toBeFalsy();
    expect(
      (component.firstChild as HTMLDListElement).classList.contains(
        "bg-primary-300"
      )
    ).toBeTruthy();
  });

  it("should match snapshot", () => {
    const { getByTestId } = render(<BubbleText>Child</BubbleText>);
    const component = getByTestId("bubble-text");

    expect(component).toMatchSnapshot();
  });
});
