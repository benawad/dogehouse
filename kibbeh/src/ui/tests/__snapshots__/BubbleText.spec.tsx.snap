// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BubbleText should match snapshot 1`] = `
<div
  class="text-primary-200 font-bold items-center"
  data-testid="bubble-text"
>
  <div
    class="inline-block mr-2 w-2 h-2 rounded-full bg-primary-300"
  />
  Child
</div>
`;
